packages/frontend/src/components/accounts/ledger/SetupLedger.js

```diff
@@ -51,6 +50,10 @@ const SetupLedger = (props) => {
     const fundingOptions = parseFundingOptions(props.location.search);
     const shouldRenderRecaptcha = !fundingOptions && RECAPTCHA_CHALLENGE_API_KEY && isNewAccount && !ENABLE_IDENTITY_VERIFIED_ACCOUNT;
 
+    const has2fa = useSelector(selectAccountHas2fa);
+    // disable the Continue button if a user has 2fa enabled, or we don't know yet if it's disabled/enabled
+    const disable2fa = has2fa || typeof has2fa === 'undefined';
```

Comment:
> The only reason to use typeof is if the variable might not be defined, but here that's not a concern.

It is, we cannot allow enabling ledger until we are sure that 2fa is not enabled - 2fa info comes with a delay.

Extended the condition so it will work properly on testnet as well.

Reply:
No, the variable _is_ defined - a variable being defined (declared) and having a value of `undefined` (uninitialized) are two different things.

What I'm saying is, the _only_ reason to use `typeof` rather than check for `undefined` directly is if the variable may not have be in scope. `typeof` ensures that referencing the variable doesn't break if it hasn't been declared. Here `has2fa` is defined with a `const` and so the `typeof` check is entirely unnecessary here.

I'm not aware of any scenario in NodeJS in which you would need to use `typeof x === 'undefined'` at all, especially when using `let` and `const` which throw `ReferenceError` when referenced before being declared.